aten/src/ATen/core/boxing/KernelFunction_impl.h

```diff
@@ -174,7 +174,7 @@ template<class FuncPtr, bool AllowLegacyTypes>
 inline KernelFunction KernelFunction::makeFromUnboxedFunction(FuncPtr func_ptr) {
     static_assert(is_compile_time_function_pointer<FuncPtr>::value, "Tried to call KernelFunction::makeFromUnboxedFunction with an invalid parameter. It must be a function pointer created with TORCH_FN.");
     static_assert(!std::is_same<typename FuncPtr::FuncType, BoxedKernelFunction>::value, "Tried to call KernelFunction::makeFromUnboxedFunction with a boxed function pointer. Please use KernelFunction::makeFromBoxedFunction instead.");
-    static_assert(FuncPtr::func_ptr() != nullptr, "Kernel function cannot be nullptr");
+    TORCH_INTERNAL_ASSERT(FuncPtr::func_ptr() != nullptr, "Kernel function cannot be nullptr");
```

Comment:
this seems like a compiler bug. why should ASAN change whether func_ptr is constexpr?

can we narrow the conditions under which we resort to performing this assertion at runtime, such as by detecting both 1) the broken compiler and 2) ASAN being enabled, and only then converting it to a TORCH_INTERNAL_ASSERT?

Also, I tried to repro this in Godbolt, and a straightforward attempt didn't seem to work: https://godbolt.org/z/MEW384qns

Reply:
Ack Done.
Please review.